python/telemetry/snet.py

```diff
@@ -168,10 +100,91 @@ def _decode(self, obj, context, path=None):
     'TargetData_ControlIsActive' / Int8ul  # flag, really
     )
 
+LTUFrameHeader = BitStruct(
+    'SrcId' / BitsInteger(7),
+    'DstId' / BitsInteger(7),
+    'FrCntTx' / BitsInteger(4),
+    'FrCntRx' / BitsInteger(4),
+    'SNR' / BitsInteger(4),
+    'AiTypeSrc' / BitsInteger(4),
+    'AiTypeDst' / BitsInteger(4),
+    'DfcId' / BitsInteger(2),
+    'Caller' / Flag,
+    'Arq' / Flag,
+    'PduTypeId' / Flag,
+    'BchRq' / Flag,
+    'Hailing' / Flag,
+    'UdFl1' / Flag,
+    'PduLength' / BitsInteger(10),
+    'CRC13' / BitsInteger(13),
+    'CRC5' / BitsInteger(5),
+    Padding(2)
+    )
+
+# For reference:
+#
+# To send mailbox messages
+# FCIDMajor = 56  (COM0)
+# FCIDSub = 34 || 134  (COM1)
+
+MailboxTelemetry = Struct(
+    'SysTime' / BitsInteger(4),
+    'MemDataType' / BitsInteger(1),
+    'MemData32' / Hex(GreedyBytes)
+    # 'MemData32' / LinearAdapter(32, Int8ub)
+    )
```

Comment:
Can you point to some documentation describing this structure? What does the field `MemData32` contain?

Reply:
There is no documentation available that I can reference, unfortunately. I got access to pieces where this was implemented, I can add more context about the values/types for each field.

Reply:
On a second look, is this actually correct? `BitsInteger(4)` is supposed to be a 4-bit integer, if I recall correctly. Then `BitsInteger(1)` is a 1-bit integer. Then we have `GreedyBytes`, but we're not byte aligned.

I guess that as it is this `Struct` is better than nothing, but it isn't very useful either. It just says that there is a timestamp, but doesn't indicate how to convert that to UTC time or whether it's seconds since boot or something, a memory data type, but without indicating what each type means, and finally a bunch of bytes of unknown content. That's why I asked for additional documentation.

Reply:
You are right on the first statement, it's an error from my side and I made a proper definition of the structure as a comment and later in the Struct. In addition to that, it shouldn't catch it as hex, the format of how you want to interpret the bits should be something outside the deframer.

As for the second point, I added more documentation from the logic I can extract, from the code I was provided. In addition to that, I fixed the links to the relevant docs of SNET at TUBerlin, however, none of them mention mailbox, and I am not aware of any documentation that explains its functionality nor implementation. I am doing my best to provide as much context with the little information I got around the topic, but I am sure the comments in the code today are accurate.